src/guides/v2.4/graphql/interfaces/configurable-product.md

```diff
@@ -738,6 +758,429 @@ The following `products` query returns `ConfigurableProduct` information about t
+        configurable_options {
+          attribute_code
+          values {
+            label
+            uid
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+**Response:**
+
+```json
+{
+  "data": {
+    "products": {
+      "items": [
+        {
+          "media_gallery": [
+            {
+              "url": "http://<host>/media/catalog/product/cache/816455256c48217ab8c5c822a6039d1a/m/s/ms10-blue_main_1.jpg",
+              "label": "",
+              "position": 1
+            },
+            {
+              "url": "http://<host>/media/catalog/product/cache/816455256c48217ab8c5c822a6039d1a/m/s/ms10-blue_alt1_1.jpg",
+              "label": "",
+              "position": 2
+            },
+            {
+              "url": "http://<host>/media/catalog/product/cache/816455256c48217ab8c5c822a6039d1a/m/s/ms10-blue_back_1.jpg",
+              "label": "",
+              "position": 3
+            }
+          ],
+          "configurable_options": [
+            {
+              "attribute_code": "color",
+              "values": [
+                {
+                  "label": "Black",
+                  "uid": "Y29uZmlndXJhYmxlLzkzLzQ5"
+                },
+                {
+                  "label": "Blue",
+                  "uid": "Y29uZmlndXJhYmxlLzkzLzUw"
+                },
+                {
+                  "label": "Red",
+                  "uid": "Y29uZmlndXJhYmxlLzkzLzU4"
+                }
+              ]
+            },
+            {
+              "attribute_code": "size",
+              "values": [
+                {
+                  "label": "XS",
+                  "uid": "Y29uZmlndXJhYmxlLzE2MC8xNjY="
+                },
+                {
+                  "label": "S",
+                  "uid": "Y29uZmlndXJhYmxlLzE2MC8xNjc="
+                },
+                {
+                  "label": "M",
+                  "uid": "Y29uZmlndXJhYmxlLzE2MC8xNjg="
+                },
+                {
+                  "label": "L",
+                  "uid": "Y29uZmlndXJhYmxlLzE2MC8xNjk="
+                },
+                {
+                  "label": "XL",
+                  "uid": "Y29uZmlndXJhYmxlLzE2MC8xNzA="
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  }
+}
+```
+
+#### Step 2. Select the color
+
+In this example, the shopper has selected the red variant. The query has been expanded to include the `configurable_product_options_selection` object, which filters on the `uid` of the red variant. Within that object, notice:
+
+-  The `options_available_for_selection` shows that all of the `size` attributes can still be selected, and that the red variant has been selected.
+-  The `media_gallery` object contains the `ms10-red_main_1.jpg` image, which can then be displayed with the parent images.
+-  The `variant` object is null, because the shopper has not selected a size. The variant is not fully defined until a value for all of the selectable attributes.
```

Comment:
```suggestion
-  The `variant` object is null, because the shopper has not selected a size. The variant is not fully defined until you specify a value for all of the selectable attributes.
```